File: docs/Design Presentation/src/fib.c
int fib(int n){
    return _fib(0, 1, n);
}

int _fib(int x0, int x1, int n){
    if(n == 0)
        return x0;
    else
        return _fib(x1, (x0 + x1), (n - 1));
}

File: H2V/tests/fib.hs
-- Test case for nested, recursive functions
-- Returns the nth fibonnacci number
-- Could be implemented using an infinite list, but that would be more painful to compile to Verilog
-- x0, x1 are accumulators, n counts down to the required index
fib n = fib' 0 1 n where
    fib' x0 _ 0 = x0
    fib' x0 x1 n = fib' x1 (x0 + x1) (n - 1)
